src/main/java/org/opensearch/knn/quantization/models/quantizationState/OneBitScalarQuantizationState.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationState;
+
+import org.opensearch.knn.quantization.models.quantizationParams.SQParams;
+import org.opensearch.knn.quantization.util.QuantizationStateSerializer;
+
+import java.io.IOException;
+
+/**
+ * OneBitScalarQuantizationState represents the state of one-bit scalar quantization,
+ * including the mean values used for quantization.
+ */
+public final class OneBitScalarQuantizationState implements QuantizationState {
+    private final SQParams quantizationParams;
+    private final float[] mean;
```

Comment:
Thoughts on renaming this variable to `meanArray`? When I see a variable named mean I would assume it's a float not a float array inherently